### ISSUE TITLE
Is it possible that MPV can be added into native player for Jellyfin Android TV?

### ISSUE THREAD

**Issue by EricChan1989** [jellyfin/jellyfin-androidtv#3007]
### Describe the feature you'd like

Dear Developer,

After using Jellyfin media player on Windows for a long while, I can see that MPV is a very stable media player and it has the best compatibilty to almost all the media formats I use daily.
On Android phone, I also use it as an external player for Jellyfin which also works perfectly.

For my Android TV, previously there was a bug for MPV which causes hardware decoding not working, but it is just fixed by the latest release.

I am just wondering that if it is possible to add MPV into native media player library in the android TV version which can has a even better support for all the functions?

The reason I would recommend this is that the VLC library is too buggy, and exoplayer is not working that perfectly as MPV.

Thanks.

**Comment by nielsvanvelzen** [comment]
This question comes up frequently and nobody has given me any good arguments as to why we should use MPV. So in short the answer is no, unless a very good reason to switch comes up.

> The reason I would recommend this is that the VLC library is too buggy

We didn't mark it experimental for fun

> and exoplayer is not working that perfectly as MPV.

Define "not working that perfectly".

**Comment by EricChan1989** [comment]
For exoplayer, the speed of loading subtiltles is very slow compared to mpv.
I can always see it says "loading subtitles" at the very beginning of the video for several seconds, and then this notification would disappear.
In that period of time, if using the seek function, the subtitle may not display.
However, in mpv, it is not the case. There is no loading notification, and also seeking to another video location can have the subtitles displayed immediately.

**Comment by nielsvanvelzen** [comment]
The subtitles aren't handled by ExoPlayer but a custom implementation (that is indeed not great), so switching the backend to MPV wouldn't fix that issue.

**Comment by EricChan1989** [comment]
Thanks for the information, then may i know if there is any possibility to get this fixed?

**Comment by nielsvanvelzen** [comment]
#145

**Comment by KyleSanderson** [comment]
Exoplayer handles this out the gate. The library call just has to be added to load the subs from there. https://github.com/moneytoo/Player is a very small implementation of ExoPlayer that has had full support for this from the start.

**Comment by Rihan9** [comment]
I tried mpv as external player and is way more laggy than exoplayer or vlc. Like about 3/4 fps total. No audio passthorught or 5.1 channels support. How can you recommend it? I'm using a different player than you? 

I'm using this: /store/apps/details?id=is.xyz.mpv

**Comment by daftblazer** [comment]
I would prefer having MPV. Anime uses .ASS subtitles very often, for both regular subs and the signs/songs track. 9 times out of 10 my shows will have to transcode, so my options are either turn off all subtitles or wait for it to transcode. I've tried using an external player, but it's not ideal at all, it doesn't seem to auto set my default language track or subtitles, and there's issues with it syncing how much I've watched.
